dist/s_all

```diff
@@ -45,95 +62,81 @@ echo "Updating files that include the package version" &&
 errchk()
 {
     if ! `test -s $2`; then
-        return
+        return 0
     fi
 
-    echo "####################### MESSAGE ############################"
-    echo "s_all run of: \"$1\" resulted in:"
-    sed -e 's/^/    /' $2
-    echo "#######################"
+    echo '#########' s_all run of: "$1" resulted in:$'\n'"$(cat $2)"$'\n''#########'
 
-    # If the test was skipped or marked warning only, ignore the failure.
-    if [ ! `grep "$1.*skipped" $2 > /dev/null` ] && [ ! "$3" = "--warning-only" ] ; then
+    if grep -q "$1.*skipped" $2 || [[ "$3" = "--warning-only" ]]; then
+        : # If the test was skipped or marked warning only, ignore the failure.
+    else
         errfound=1;
+        echo "$1" >> "${t_pfx}_errfound"
     fi
 
     rm -f $2
+    return $errfound
 }
 
 run()
 {
-    2>&1 $1 > $t
-    errchk "$1" $t "$2"
+    local cmd="$1" flags="$2"
+    local name="${cmd%% *}"
+    local t="${t_pfx}-${name}-$$"
+    [[ "$name" =~ .*\.py$ ]] && cmd="python3 $cmd" || cmd="/bin/bash $cmd"
+    $cmd > $t 2>&1
```

Comment:
This is stylistic so feel free to disregard, but I find 
```
if [[ "$name" =~ .*\.py$ ]] ; then
    python3 $cmd > $t 2>&1
else
    /bin/bash $cmd > $t 2>&1
fi
```
more readable than `&&` and `||`

Reply:
I didn't want to execude the `cmd` right there because there's chance that there will be more processing to it, like echoing the command executed. I'd like to keep it in a variable and execute uniformly. This syntax `A && B || C` is bash is pretty much like if-then-else or ternary operator. If it's not readable well enough, let's think of amother way to express it.

Reply:
No worries :) it's stylistic so happy to resolve the convo as it 👍